met-web/src/components/LandingPage/LandingPage.tsx

```diff
@@ -5,28 +5,69 @@ import Grid from '@mui/material/Grid';
 import { Link } from 'react-router-dom';
 import { MetPageGridContainer, PrimaryButton } from '../common';
 import { Engagement } from 'models/engagement';
-import { useAppSelector, useAppDispatch } from 'hooks';
-import { HeadCell } from '../common/Table/types';
+import { useAppDispatch } from 'hooks';
+import { HeadCell, PageInfo, PaginationOptions } from '../common/Table/types';
 import { formatDate } from 'components/common/dateHelper';
 import { Link as MuiLink } from '@mui/material';
-import { fetchAll } from 'services/engagementService';
+import { getEngagements } from 'services/engagementService';
 import SearchIcon from '@mui/icons-material/Search';
 import Stack from '@mui/material/Stack';
+import { openNotification } from 'services/notificationService/notificationSlice';
 
 const LandingPage = () => {
     const [searchFilter, setSearchFilter] = useState({
         key: 'name',
         value: '',
     });
     const [searchText, setSearchText] = useState('');
+    const [engagements, setEngagements] = useState<Engagement[]>([]);
+    const [paginationOptions, setPaginationOptions] = useState<PaginationOptions<Engagement>>({
+        page: 1,
+        size: 10,
+        sort_key: 'name',
+        nested_sort_key: null,
+        sort_order: 'asc',
+    });
+
+    const [pageInfo, setPageInfo] = useState<PageInfo>({
+        total: 0,
+    });
+    const [tableLoading, setTableLoading] = useState(true);
 
     const dispatch = useAppDispatch();
 
+    const { page, size, sort_key, nested_sort_key, sort_order } = paginationOptions;
+
     useEffect(() => {
-        fetchAll(dispatch);
-    }, [dispatch]);
+        callGetEngagements();
+    }, [paginationOptions, searchFilter]);
 
-    const rows = useAppSelector<Engagement[]>((state) => state.engagement.allEngagements);
+    const callGetEngagements = async () => {
```

Comment:
The method does more than just call the get engagements, maybe load engagements makes more sense?

Reply:
agreed, thank you